string state
---
int8 errorValue
